platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/SamlValidator.java

```diff
@@ -0,0 +1,416 @@
+            throw new ValidationException("Signature validation failed.", e);
+        }
+    }
+
+    public static class Builder {
+        protected SimpleSign simpleSign;
+
+        protected SamlProtocol.Binding binding;
+
+        protected boolean isRequest;
+
+        protected XMLObject xmlObject;
+
+        protected Duration issueTimeout = Duration.ofMinutes(10);
+
+        /*Should we be handling jitter differently?*/
+        protected Duration jitter = Duration.ofSeconds(30);
+
+        protected String inResponse;
+
+        protected String endpoint;
+
+        protected String relayState;
+
+        protected String signature;
+
+        protected String sigAlgo;
+
+        protected String samlString;
+
+        protected String signingCertificate;
+
+        public Builder(SimpleSign simpleSign) {
+            this.simpleSign = simpleSign;
+        }
+
+        public void buildAndValidate(@NotNull String destination,
+                @NotNull SamlProtocol.Binding binding, @NotNull ValidatingXMLObject xmlObject)
+                throws IllegalStateException, ValidationException {
+            Validator<ValidatingXMLObject> validator = build(destination, binding, xmlObject);
+            xmlObject.registerValidator(validator);
+            xmlObject.validate(false);
+        }
+
+        /**
+         * @param endpoint
+         * @param binding
+         * @param xmlObject target object to validate
+         * @return
+         * @throws IllegalStateException
+         */
+        public SamlValidator build(@NotNull String endpoint, @NotNull SamlProtocol.Binding binding,
+                @NotNull ValidatingXMLObject xmlObject)
+                throws IllegalStateException, ValidationException {
+            if (binding == null) {
+                throw new IllegalArgumentException("Binding cannot be null!");
+            }
+            this.binding = binding;
+
+            if (isBlank(endpoint)) {
+                throw new IllegalArgumentException("The service endpoint destination cannot be null");
+            }
+            this.endpoint = endpoint;
+
+            if (xmlObject instanceof LogoutRequest) {
+                isRequest = true;
+                LOGGER.trace("xmlObject is a LogoutRequest [{}]", xmlObject);
+            } else if (xmlObject instanceof LogoutResponse) {
+                isRequest = false;
+                LOGGER.trace("xmlObject is a LogoutResponse [{}]", xmlObject);
+            } else {
+                throw new IllegalArgumentException("Could not determine type of xmlObject");
+            }
+            this.xmlObject = xmlObject;
+
+            Validator<ValidatingXMLObject> validator = null;
+            if (binding == SamlProtocol.Binding.HTTP_POST) {
+                return isRequest ? new PostRequest(this) : new PostResponse(this);
+            }
+            if (binding == SamlProtocol.Binding.HTTP_REDIRECT) {
```

Comment:
Same comment here. We shouldn't be validating stuff in security-core-impl that depends on implementations that may or may not exist at a lower level in the hierarchy.